src/_pytest/assertion/util.py

```diff
@@ -134,6 +134,13 @@ def isiterable(obj: Any) -> bool:
         return not istext(obj)
     except TypeError:
         return False
+    except Exception as e:
+        raise ValueError(
```

Comment:
I think we should probably just replace `except TypeError:` with `except Exception:` here - if calling `iter(obj)` raises an error, it seems reasonable to treat that as "not iterable" rather than a special error case.

Reply:
Hm I think the error message is helpful if people don't expect it, but I can see that this adds to things to maintain